### PR TITLE
Teet 165 - appointment

### PR DESCRIPTION
**Pull request by trongnhandev1011**

### CODE REVIEW

File: src/containers/ExaminationScreen/ExaminationScreen.container.jsx [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
   const [showInfo, setShowInfo] = useState(false);
+  const [bookingArray, setBookingArray] = useState();
   const { bookingId } = useParams();
   const [isRendered, setIsRendered] = useState(false);
-
+  const history = useHistory();
   const [bookingData, setBookingData] = useState({});
 
   const [form] = Form.useForm();
+  const index = bookingArray ? bookingArray.indexOf(bookingId) : -1;
```
Comment (ZERO-94) [inline src/containers/ExaminationScreen/ExaminationScreen.container.jsx:32]: Please comment in here, why do you have this ternary?

File: src/containers/ExaminationScreen/ExaminationScreen.container.jsx [hunk 2]
```diff
@@ -30,12 +47,72 @@ const ExaminationScreenContainer = () => {
...
+
+  const returnToDashboard = () => {
+    history.push(RoutePath.DASHBOARD_PAGE);
+  };
+
   useEffect(() => {
     bookingId && fetchBookingData();
   }, [bookingId]);
 
+  useEffect(() => {
```
Comment (ZERO-94) [inline src/containers/ExaminationScreen/ExaminationScreen.container.jsx:86]: 2 useEffect with the same dependency????

File: src/containers/ExaminationScreen/ExaminationScreen.container.jsx [hunk 3]
```diff
@@ -30,12 +47,72 @@ const ExaminationScreenContainer = () => {
...
+        bookingData?.confirmed &&
+        index !== bookingArray.length ? (
+          <PageHeader
+            ghost={false}
+            title="Next examination:"
+            subTitle={`BookingId: ${bookingArray[index + 1]?.id}`}
+          >
+            <Descriptions size="small" column={3}>
+              <Descriptions.Item label="Next patient">
+                {bookingArray[index + 1]?.patient?.firstName +
```
Comment (ZERO-94) [inline src/containers/ExaminationScreen/ExaminationScreen.container.jsx:103]: Can you get the first booking before return the component?  Too much "."

File: src/containers/UpdateBookingForm/UpdateBookingForm.container.jsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import { updateBookingDuringTreatment } from "../../services/teeth-apis/BookingController";
+import RequestUpdateFormToRequestUpdateData from "../../mapper/RequestUpdateFormToRequestUpdateData.js";
+import ServicePickingModalContainer from "../ServicePickingModal/ServicePickingModal.container";
+import { notification } from "antd";
+
+const UpdateBookingFormContainer = ({
+  isUpdated,
+  bookingId,
+  setIsRendered,
+  setShowInfo,
```
Comment (ZERO-94) [inline src/containers/UpdateBookingForm/UpdateBookingForm.container.jsx:12]: Why do you need setShowInfo

File: src/services/teeth-apis/AppointmentController.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { CallAPI } from "../teeth-apis/axiosBase";
+import { APPOINTMENT_END_POINT } from "../end-points/AppointmentEndPoints";
+
+export const getAllAppointments = (options) =>
+  CallAPI(`${APPOINTMENT_END_POINT}`, "GET", {}, options);
+
+export const getAppointmentById = (appointmentId) =>
+  CallAPI(`${APPOINTMENT_END_POINT}/${appointmentId}`, "GET");
+
+export const createAppointments = (values) =>
```
Comment (ZERO-94) [inline src/services/teeth-apis/AppointmentController.js:10]: Appointment data is better than values?

**Review by ZERO-94 — CHANGES_REQUESTED** [review]

File: src/containers/UpdateBookingForm/UpdateBookingForm.container.jsx (hunk 4 above)
Comment (trongnhandev1011) [reply src/containers/UpdateBookingForm/UpdateBookingForm.container.jsx:12]: To display the alert when waiting for confirm update. This problem is from the previous TEET and you said we could use useRef to handle it but this solution's just fine

File: src/components/DentistUpdateBookingForm/DentistUpdateBookingForm.component.jsx [hunk 6]
```diff
@@ -29,13 +29,19 @@ const DentistUpdateBookingFormComponent = ({
   return (
     <Col span={12} style={{ margin: "30px 40px" }}>
       {bookingData?.status === BookingStatusConstants.TREATMENT ? (
-        <Button
-          style={{ marginBottom: 20 }}
-          onClick={() => switchUpdateState()}
-        >
-          Update
-        </Button>
+        <div style={{ display: "flex", justifyContent: "space-between" }}>
```
[the remark this answers is not in the document]
Comment (trongnhandev1011) [reply src/components/DentistUpdateBookingForm/DentistUpdateBookingForm.component.jsx:32]: for easily styling

File: src/containers/AppointmentDetailModal/AppointmentDetailModal.container.jsx [hunk 7]
```diff
@@ -55,31 +53,22 @@ const AppointmentDetailModalContainer = ({
...
-      >
-        <Form name="create_booking_from_modal_form" onFinish={onFinish}>
+  const createBooking = () => {
+    Modal.info({
+      title: "Register for the next examination",
+      maskClosable: true,
+      closable: true,
+      okButtonProps: { style: { display: "none" } },
+      content: (
+        <Form
```
Comment (ZERO-94) [inline src/containers/AppointmentDetailModal/AppointmentDetailModal.container.jsx:67]: this should be put in a separate component (function)

File: src/containers/ExaminationScreen/ExaminationScreen.container.jsx [hunk 8]
```diff
@@ -1,35 +1,21 @@
...
-  const [bookingArray, setBookingArray] = useState();
   const { bookingId } = useParams();
   const [isRendered, setIsRendered] = useState(false);
   const history = useHistory();
   const [bookingData, setBookingData] = useState({});
 
   const [form] = Form.useForm();
-  const index = bookingArray ? bookingArray.indexOf(bookingId) : -1;
+
+  //get the index of the first booking that is not confirmed
```
Comment (ZERO-94) [inline src/containers/ExaminationScreen/ExaminationScreen.container.jsx:18]: comment no code ???

**Review by ZERO-94 — CHANGES_REQUESTED** [review]

**Review by ZERO-94 — APPROVED** [review]

**Pull request merged**